src/transformers/file_utils.py

```diff
@@ -306,10 +311,14 @@ def http_get(url, temp_file, proxies=None, resume_size=0, user_agent=None):
 
 def get_from_cache(
     url, cache_dir=None, force_download=False, proxies=None, etag_timeout=10, resume_download=False, user_agent=None
-):
+) -> Optional[str]:
     """
-    Given a URL, look for the corresponding dataset in the local cache.
+    Given a URL, look for the corresponding file in the local cache.
     If it's not there, download it. Then return the path to the cached file.
+
+    Return:
+        None in case of non-recoverable file (non-existent or inaccessible url + no cache on disk).
+        Local path (string) otherwise
```

Comment:
@thomwolf @LysandreJik This is mostly a refactor of the logic keeping the existing behavior of `get_from_cache`. The logic flow is easier to understand now, IMO.

The only behavior change is when etag is None, the cached file doesn't exist, AND there are no matching files on disk: we used to return the cached file (which doesn't exist). We now return None and document this behavior.

Also cc @aaugustin as he refactored this same method pretty recently.

Reply:
Nice!